[es]: Agrega traduccion de files/en-us/learn_web_development/getting_started/web_standards/how_browsers_load_websites (#26303)
mdn/translated-content: files/es/learn_web_development/getting_started/web_standards/how_browsers_load_websites/index.md

---
title: Cómo los navegadores cargan los sitios web
slug: Learn_web_development/Getting_started/Web_standards/How_browsers_load_websites
l10n:
  sourceCommit: bca4bdeae2c1e3e673c3c138b7003220b164596e
---

{{LearnSidebar}}

{{PreviousMenuNext("Learn_web_development/Getting_started/Web_standards/The_web_standards_model", "Learn_web_development/Getting_started/Soft_skills", "Learn_web_development/Getting_started/Web_standards")}}

En el artículo anterior, vimos una [visión general de las tecnologías](/es/docs/Learn_web_development/Getting_started/Web_standards/The_web_standards_model#overview_of_modern_web_technologies) con las que se construyen los sitios web. En este artículo, recorreremos el proceso mediante el cual se renderizan esas tecnologías: cuando un navegador ha recibido los archivos de código y otros recursos que componen una página web (como se explicó en [Cómo funciona la web](/es/docs/Learn_web_development/Getting_started/Web_standards/How_the_web_works)), ¿cómo se juntan para crear la experiencia final con la que interactúa el usuario?

<table>
  <tbody>
    <tr>

>       <th scope="row">Requisitos Previos:</th>

Reviewer: ```suggestion
      <th scope="row">Requisitos previos:</th>
```

- Elementos {{htmlelement("script")}} que hacen referencia a archivos [JavaScript](/es/docs/Learn_web_development/Core/Scripting) externos.
- Elementos como {{htmlelement("img")}}, {{htmlelement("video")}} y {{htmlelement("audio")}}, que hacen referencia a archivos multimedia que deseas incrustar en la página web.

## Análisis de CSS y renderización de la página

A continuación, veamos cómo se maneja el CSS.

1. El navegador analiza el CSS que se encuentra en la página (ya sea incluido en el archivo HTML o obtenido de hojas de estilo externas) y clasifica las diferentes reglas de estilo CSS en diferentes "contenedores" según a qué elementos HTML (representados en el DOM como elementos llamados **nodos**) se aplicarán. Luego, el navegador adjunta estilos a los diferentes elementos según sea necesario (este paso intermedio se llama árbol de renderizado).
2. El árbol de renderizado se presenta en la estructura en la que debería aparecer después de que se hayan aplicado las reglas. Esto incluye cualquier imagen y otros archivos multimedia que se van a incrustar en la página.

> 3. La visualización visual de la página se muestra en la pantalla (esta etapa se llama pintura).

Reviewer: ```suggestion
3. La visualización visual de la página se muestra en la pantalla (esta etapa se llama pintar).
```

El siguiente diagrama ofrece una visualización del proceso del que hemos hablado hasta ahora:

![Proceso de renderización](rendering.svg)

Volviendo a nuestro ejemplo, digamos que el siguiente CSS se encuentra en el archivo HTML:

```html hidden
<p>
  Usemos:
  <span>HTML</span>
  <span>CSS</span>
  <span>JavaScript</span>
</p>
```

```css
span {
  border: 1px solid black;
  background-color: lime;
}
```

La única regla disponible en el CSS tiene un selector `span`, ¡así que el navegador puede ordenar el CSS muy rápidamente! Aplica esa regla a cada uno de los tres nodos SPAN en el árbol DOM, dándoles un borde negro y un fondo verde lima, luego pinta la representación visual final en la pantalla.

La salida actualizada es la siguiente:

{{EmbedLiveSample('Parsing the CSS, and rendering the page', '100%', 90)}}

## Manejo de JavaScript

> Cualquier JavaScript que se encuentre en la página (ya sea incluido en el archivo HTML o obtenido de archivos de script externos) se analiza, interpreta, compila y ejecuta. Esto sucede en algún momento antes de que se complete la renderización final de la página; después de todo, algún JavaScript puede afectar la renderización, por ejemplo, agregando nodos al DOM o modificando los existentes.

Reviewer: ```suggestion
Cualquier JavaScript que se encuentre en la página (ya sea incluido en el archivo HTML u obtenido de archivos de script externos) se analiza, interpreta, compila y ejecuta. Esto sucede en algún momento antes de que se complete la renderización final de la página; después de todo, algún JavaScript puede afectar la renderización, por ejemplo, agregando nodos al DOM o modificando los existentes.
```

Aprenderás más sobre esto más adelante, en nuestro módulo de [Accesibilidad](/es/docs/Learn_web_development/Core/Accessibility).

## El navegador: un entorno de programación hostil _e_ increíble

El desarrollo web front-end a veces puede ser frustrante, y algunas personas consideran que el navegador es un entorno de programación hostil. Esto se debe a que, a diferencia de otros entornos de programación, es mucho más difícil hacer garantías sobre el entorno en el que se ejecutará tu código. No puedes saber de antemano todas las diferentes combinaciones de sistema operativo, navegador, idioma, ubicación, conexión de red, CPU, GPU, memoria, duración de la batería, etc., que tendrán tus usuarios, por lo tanto, no puedes garantizar una experiencia de usuario perfecta para todos ellos.

> Los navegadores modernos tienden a implementar los estándares web de manera bastante consistente, pero todavía hay mucha incertidumbre que sortear. Como desarrollador web, deberás aceptar esa incertidumbre, programar a la defensiva y ser conservador con las características que uses. Esto se basa en adherirse a las [mejores prácticas](/es/docs/Learn_web_development/Getting_started/Web_standards/The_web_standards_model#web_best_practices) descritas en el artículo anterior.

Reviewer: No estoy muy seguro de que la palabra se entienda como tal, en todos los países hispanohablantes

```suggestion
Los navegadores modernos tienden a implementar los estándares web de manera bastante consistente, pero todavía hay mucha incertidumbre por navegar. Como desarrollador web, deberás aceptar esa incertidumbre, programar a la defensiva y ser conservador con las características que uses. Esto se basa en adherirse a las [mejores prácticas](/es/docs/Learn_web_development/Getting_started/Web_standards/The_web_standards_model#web_best_practices) descritas en el artículo anterior.
```
